### PR TITLE
8230841: Remove oopDesc::equals()

### PR DESCRIPTION
**Pull request by shipilev**
This is the second patch in the series to cleanup/revert the GC BarrierSets to most current state. See the dependent ("this issue blocks" links) issues to see what is next.

The equals APIs were added for Shenandoah GC, in the incarnation that exposed both from- and to-copies to the runtime system (see JDK-8199781). Since then, Shenandoah greatly simplified this by switching to a strong to-space invariant, which obviates the need for this API. Therefore, it was removed in JDK 14. This patch reverts JDK 11 parts to that shape as well, which should simplify 11u maintenance.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8230841](https://bugs.openjdk.java.net/browse/JDK-8230841): Remove oopDesc::equals()

### Reviewers
 * [Roman Kennke](https://openjdk.java.net/census#rkennke) (@rkennke - Committer)
 * [Christoph Langer](https://openjdk.java.net/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/204/head:pull/204` \
`$ git checkout pull/204`

Update a local copy of the PR: \
`$ git checkout pull/204` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/204/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 204`

View PR using the GUI difftool: \
`$ git pr show -t 204`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/204.diff">https://git.openjdk.java.net/jdk11u-dev/pull/204.diff</a>

</details>

### CODE REVIEW

**Comment by shipilev** [comment]
Paging @rkennke to sanity check this.

**Review by rkennke — APPROVED** [review]
LGTM. One small comment, I leave this up to you. Thanks!

File: src/hotspot/share/classfile/systemDictionary.cpp
Comment (shipilev) [reply src/hotspot/share/classfile/systemDictionary.cpp:521]: Thanks! It is like that in the upstream version (16u), so I would leave it as is.

**Review by RealCLanger — APPROVED** [review]

**Comment by shipilev** [comment]
Got the push approval.

/integrate

**Pull request closed without merging**
